#! /bin/sh /usr/share/dpatch/dpatch-run
## build_w_camlp5.dpatch by Stefano Zacchiroli <zack@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Enable building against camlp5 (the "classical" version of camlp4)

@DPATCH@
diff -urNad trunk~/Makefile.rules trunk/Makefile.rules
--- trunk~/Makefile.rules	2007-07-31 18:20:57.000000000 +0200
+++ trunk/Makefile.rules	2007-09-08 12:42:33.000000000 +0200
@@ -13,7 +13,7 @@
 OCAMLFIND = ocamlfind
 OCAMLYACC = ocamlyacc
 OCAMLLEX  = ocamllex
-CAMLP4    = camlp4
+CAMLP4    = camlp5
 OCAMLMKLIB = ocamlmklib
 OCAMLDOC  = $(OCAMLFIND) ocamldoc
 
diff -urNad trunk~/src/equeue-gtk1/Makefile trunk/src/equeue-gtk1/Makefile
--- trunk~/src/equeue-gtk1/Makefile	2007-07-31 18:20:59.000000000 +0200
+++ trunk/src/equeue-gtk1/Makefile	2007-09-08 12:42:33.000000000 +0200
@@ -9,7 +9,7 @@
 
 INCLUDES += $(INC_EQUEUE)
 
-PP_OPTIONS = -syntax camlp4o -package camlp4.macro $(GTK_DEFINES)
+PP_OPTIONS = -syntax camlp5o -package camlp5.macro $(GTK_DEFINES)
 
 include $(TOP_DIR)/Makefile.rules
 
diff -urNad trunk~/src/equeue-gtk1/Makefile.pre trunk/src/equeue-gtk1/Makefile.pre
--- trunk~/src/equeue-gtk1/Makefile.pre	2007-07-31 18:20:59.000000000 +0200
+++ trunk/src/equeue-gtk1/Makefile.pre	2007-09-08 12:42:33.000000000 +0200
@@ -6,6 +6,6 @@
 
 include ./Makefile.gtk
 
-PP_OPTIONS = -syntax camlp4o -package camlp4.macro $(GTK_DEFINES)
+PP_OPTIONS = -syntax camlp5o -package camlp5.macro $(GTK_DEFINES)
 
 include $(TOP_DIR)/Makefile.rules
diff -urNad trunk~/src/equeue-gtk2/Makefile.pre trunk/src/equeue-gtk2/Makefile.pre
--- trunk~/src/equeue-gtk2/Makefile.pre	2007-07-31 18:20:59.000000000 +0200
+++ trunk/src/equeue-gtk2/Makefile.pre	2007-09-08 12:42:33.000000000 +0200
@@ -6,7 +6,7 @@
 
 include ./Makefile.gtk
 
-PP_OPTIONS = -syntax camlp4o -package camlp4.macro $(GTK_DEFINES)
+PP_OPTIONS = -syntax camlp5o -package camlp5.macro $(GTK_DEFINES)
 
 include $(TOP_DIR)/Makefile.rules
 
diff -urNad trunk~/src/nethttpd-for-netcgi1/Makefile trunk/src/nethttpd-for-netcgi1/Makefile
--- trunk~/src/nethttpd-for-netcgi1/Makefile	2007-07-31 18:20:59.000000000 +0200
+++ trunk/src/nethttpd-for-netcgi1/Makefile	2007-09-08 12:42:33.000000000 +0200
@@ -8,7 +8,7 @@
 	  nethttpd_engine.mli nethttpd_services.mli nethttpd_plex.mli
 
 INCLUDES += $(INC_EQUEUE) $(INC_NETSTRING) $(INC_NETCGI1) $(INC_NETPLEX)
-PP_OPTIONS = -package "camlp4, camlp4.macro" -syntax "camlp4o"
+PP_OPTIONS = -package "camlp5, camlp5.macro" -syntax "camlp5o"
 
 #PP_OPTIONS_FINDLIB += -ppopt "-DTesting"
 # do this to enable test coverage calculations
diff -urNad trunk~/src/nethttpd-for-netcgi1/Makefile.pre trunk/src/nethttpd-for-netcgi1/Makefile.pre
--- trunk~/src/nethttpd-for-netcgi1/Makefile.pre	2007-09-08 12:42:32.000000000 +0200
+++ trunk/src/nethttpd-for-netcgi1/Makefile.pre	2007-09-08 12:42:33.000000000 +0200
@@ -6,6 +6,6 @@
 
 INSTALL_EXTRA +=
 
-PP_OPTIONS = -package "camlp4, camlp4.macro" -syntax "camlp4o"
+PP_OPTIONS = -package "camlp5, camlp5.macro" -syntax "camlp5o"
 
 include $(TOP_DIR)/Makefile.rules
diff -urNad trunk~/src/nethttpd-for-netcgi1/nethttpd_types.mli trunk/src/nethttpd-for-netcgi1/nethttpd_types.mli
--- trunk~/src/nethttpd-for-netcgi1/nethttpd_types.mli	2007-07-31 18:20:59.000000000 +0200
+++ trunk/src/nethttpd-for-netcgi1/nethttpd_types.mli	2007-09-08 12:43:01.000000000 +0200
@@ -117,10 +117,10 @@
    *)
 
 class redirected_environment : 
-        ?in_state : Netcgi1_compat.Netcgi_env.input_state ->
-        ?in_header : http_header ->
-        ?properties : (string * string) list ->
-        ?in_channel : Netchannels.in_obj_channel ->
+        ?in_state: Netcgi1_compat.Netcgi_env.input_state ->
+        ?in_header: http_header ->
+        ?properties: (string * string) list ->
+        ?in_channel: Netchannels.in_obj_channel ->
         extended_environment ->
           extended_environment
   (** This class overlays the input-side containers of an existing environment.
diff -urNad trunk~/src/nethttpd-for-netcgi2/Makefile trunk/src/nethttpd-for-netcgi2/Makefile
--- trunk~/src/nethttpd-for-netcgi2/Makefile	2007-07-31 18:20:59.000000000 +0200
+++ trunk/src/nethttpd-for-netcgi2/Makefile	2007-09-08 12:42:33.000000000 +0200
@@ -9,7 +9,7 @@
 	  nethttpd_intro.txt
 
 INCLUDES += $(INC_EQUEUE) $(INC_NETSTRING) $(INC_NETCGI2) $(INC_NETPLEX)
-PP_OPTIONS = -package "camlp4, camlp4.macro" -syntax "camlp4o"
+PP_OPTIONS = -package "camlp5, camlp5.macro" -syntax "camlp5o"
 
 #PP_OPTIONS_FINDLIB += -ppopt "-DTesting"
 # do this to enable test coverage calculations
diff -urNad trunk~/src/nethttpd-for-netcgi2/Makefile.pre trunk/src/nethttpd-for-netcgi2/Makefile.pre
--- trunk~/src/nethttpd-for-netcgi2/Makefile.pre	2007-09-08 12:42:32.000000000 +0200
+++ trunk/src/nethttpd-for-netcgi2/Makefile.pre	2007-09-08 12:42:33.000000000 +0200
@@ -15,7 +15,7 @@
 
 INSTALL_EXTRA +=
 
-PP_OPTIONS = -package "camlp4, camlp4.macro" -syntax "camlp4o"
+PP_OPTIONS = -package "camlp5, camlp5.macro" -syntax "camlp5o"
 
 include $(TOP_DIR)/Makefile.rules
 
diff -urNad trunk~/src/netplex/Makefile trunk/src/netplex/Makefile
--- trunk~/src/netplex/Makefile	2007-07-31 18:21:00.000000000 +0200
+++ trunk/src/netplex/Makefile	2007-09-08 12:42:33.000000000 +0200
@@ -19,7 +19,7 @@
 ALLMT_EXTRA = netplex_mt.cmo
 OPTMT_EXTRA = netplex_mt.cmx
 
-PP_OPTIONS = -syntax camlp4o -package camlp4
+PP_OPTIONS = -syntax camlp5o -package camlp5
 
 OCAMLC_OPTIONS_FOR_netplex_mt.ml = $(THREAD)
 OCAMLOPT_OPTIONS_FOR_netplex_mt.ml = $(THREAD)
diff -urNad trunk~/src/netplex/Makefile.pre trunk/src/netplex/Makefile.pre
--- trunk~/src/netplex/Makefile.pre	2007-07-31 18:21:00.000000000 +0200
+++ trunk/src/netplex/Makefile.pre	2007-09-08 12:42:33.000000000 +0200
@@ -9,7 +9,7 @@
 INSTALL_EXTRA = netplex-packlist netplex_mt.cmo netplex_mt.cmx netplex_mt.o
 INSTOTHER = install-netplex-admin
 
-PP_OPTIONS = -syntax camlp4o -package camlp4
+PP_OPTIONS = -syntax camlp5o -package camlp5
 
 include $(TOP_DIR)/Makefile.rules
 
diff -urNad trunk~/src/netplex/netplex_config.ml trunk/src/netplex/netplex_config.ml
--- trunk~/src/netplex/netplex_config.ml	2007-07-31 18:21:00.000000000 +0200
+++ trunk/src/netplex/netplex_config.ml	2007-09-08 12:42:33.000000000 +0200
@@ -29,7 +29,7 @@
   and semi_parse_tree_list =
     parser
       | [< 'Kwd ";";
-	   tl = parse_tree_list;
+	   tl = parse_tree_list
 	>] -> tl
       | [< >] -> []
 	  
@@ -47,7 +47,7 @@
   and parse_rhs =
     parser
       | [< 'Kwd "=";
-	   v = parse_param_value;
+	   v = parse_param_value
 	>] ->
 	  `Parameter v
       | [< 'Kwd "{";
diff -urNad trunk~/src/netstring/Makefile.def trunk/src/netstring/Makefile.def
--- trunk~/src/netstring/Makefile.def	2007-07-31 18:20:58.000000000 +0200
+++ trunk/src/netstring/Makefile.def	2007-09-08 12:42:33.000000000 +0200
@@ -46,7 +46,7 @@
 	netchannels_tut.txt netmime_tut.txt netsendmail_tut.txt \
 	netulex_tut.txt neturl_tut.txt
 
-PP_OPTIONS = -package camlp4 -syntax camlp4o
+PP_OPTIONS = -package camlp5 -syntax camlp5o
 
 OCAMLC_OPTIONS_FOR_netstring_mt.ml = $(THREAD)
 OCAMLOPT_OPTIONS_FOR_netstring_mt.ml = $(THREAD)
diff -urNad trunk~/src/netstring/netmime.ml trunk/src/netstring/netmime.ml
--- trunk~/src/netstring/netmime.ml	2007-07-31 18:20:58.000000000 +0200
+++ trunk/src/netstring/netmime.ml	2007-09-08 12:42:33.000000000 +0200
@@ -579,7 +579,7 @@
 ;;
 
 
-let storage ?ro ?fin : store -> (mime_body * out_obj_channel) = function
+let storage ?ro ?fin = function
     `Memory ->
       let body = new memory_mime_body_int ~ro_first:false ?ro "" in
       let body_ch = body#open_value_wr() in
